File: 643-maximum-average-subarray-i/maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        for (int i = 1; i < nums.length; i++)
            nums[i] += nums[i - 1];
        double ans = (double) nums[k - 1] / k;
        for (int i = 1; i <= nums.length - k; i++)
            ans = Math.max(ans, ((double) nums[k + i - 1] - nums[i - 1]) / k);
        return ans;
    }
}

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        int sum = 0, ans = 0;
        for(int i  = 0;i<k;++i)
            sum += nums[i]; 
        ans = sum;
        for(int i = k;i<nums.size();++i){
            sum = sum + nums[i] - nums[i - k];
            ans = max(ans, sum);
        }
        return (double)ans/k;
    }
};